--- deprectated/utils/gtk/lqt_gtk.c	2011-01-16 17:28:08.762319403 +0100
+++ good/utils/gtk/lqt_gtk.c	2011-01-16 17:28:40.529319398 +0100
@@ -332,10 +332,10 @@ lqtgtk_create_parameter_widget(lqt_param
       ret->label = gtk_label_new(TR_DOM(info->real_name));
       gtk_misc_set_alignment(GTK_MISC(ret->label), 0.0, 0.5);
 
-      ret->widget = gtk_combo_box_new_text();
+      ret->widget = gtk_combo_box_text_new();
       for(i = 0; i < info->num_stringlist_options; i++)
         {
-	gtk_combo_box_append_text (GTK_COMBO_BOX (ret->widget), 
+	gtk_combo_box_text_append_text (GTK_COMBO_BOX_TEXT (ret->widget), 
 				   info->stringlist_options[i]);
         }
       g_signal_connect(GTK_COMBO_BOX (ret->widget),
