diff --git a/src/util/thread.h b/src/util/thread.h
index 949bb42..687c8ad 100644
--- a/src/util/thread.h
+++ b/src/util/thread.h
@@ -24,6 +24,7 @@
 #include <jthread.h>
 #include <jmutex.h>
 #include <jmutexautolock.h>
+#include "event.h"
 
 template<typename T>
 class MutexedVariable
diff --git a/src/util/event.h b/src/util/event.h
new file mode 100644
index 0000000..571720d
--- /dev/null
+++ b/src/util/event.h
@@ -0,0 +1,51 @@
+#pragma once
+
+#ifdef _WIN32
+
+class Event {
+	HANDLE hEvent;
+
+public:
+	Event() {
+		hEvent = CreateEvent(NULL, 0, 0, NULL);
+	}
+	
+	~Event() {
+		CloseHandle(hEvent);
+	}
+	
+	void wait() {
+		WaitForSingleObject(hEvent, INFINITE);
+	}
+	
+	void signal() {
+		SetEvent(hEvent);
+	}
+};
+
+#else
+
+#include <semaphore.h>
+
+class Event {
+	sem_t sem;
+
+public:
+	Event() {
+		sem_init(&sem, 0, 0);
+	}
+	
+	~Event() {
+		sem_destroy(&sem);
+	}
+	
+	void wait() {
+		sem_wait(&sem);
+	}
+	
+	void signal() {
+		sem_post(&sem);
+	}
+};
+
+#endif
