diff --git a/boost-1.55.0/boost/fusion/container/map/detail/map_impl.hpp b/boost-1.55.0/boost/fusion/container/map/detail/map_impl.hpp
index cea2549..b03cd4e 100644
--- a/boost/fusion/container/map/detail/map_impl.hpp
+++ b/boost/fusion/container/map/detail/map_impl.hpp
@@ -73,7 +73,7 @@ namespace boost { namespace fusion { namespace detail
         {}
 
         map_impl(map_impl&& rhs)
-          : rest_type(std::forward<rest_type>(*static_cast<rest_type*>(this)))
+          : rest_type(std::forward<rest_type>(rhs.get_base()))
           , element(std::forward<Pair>(rhs.element))
         {}
 
