diff --git a/boost/python/detail/wrap_python.hpp b/boost/python/detail/wrap_python.hpp
index eaef784..ba02516 100644
--- a/boost/python/detail/wrap_python.hpp
+++ b/boost/python/detail/wrap_python.hpp
@@ -136,6 +136,14 @@ typedef int pid_t;
 
 #endif // _WIN32
 
+#ifndef SIZEOF_LONG
+# if defined(__x86_64__)
+#  define SIZEOF_LONG 8
+# else  // __x86_64__
+#  define SIZEOF_LONG 4
+# endif // __x86_64__
+#endif // SIZEOF_LONG
+
 #if PY_MAJOR_VERSION == 2 && PY_MINOR_VERSION == 2 && PY_MICRO_VERSION < 2
 # include <boost/python/detail/python22_fixed.h>
 #else
