--- Modules/FindBoost.cmake	2013-06-10 16:54:55.000000000 +0400
+++ Modules/FindBoost.cmake	2013-07-19 23:04:11.785055537 +0400
@@ -906,6 +906,7 @@
   if(${COMPONENT} STREQUAL "python" OR ${COMPONENT} STREQUAL "mpi_python")
     # Get version of Python.
     execute_process(COMMAND python -c "import sys; sys.stdout.write('.'.join(str(x) for x in sys.version_info[:2]))" OUTPUT_VARIABLE _python_version)
+    string(REGEX REPLACE "^([0-9]+)\\..*" "\\1" _python_major_version "${_python_version}")
   endif()
 
 
@@ -924,7 +925,12 @@
         ${Boost_LIB_PREFIX}boost_${COMPONENT}-${_python_version}${_boost_COMPILER}${_boost_MULTITHREADED}${_boost_RELEASE_ABI_TAG}
         ${Boost_LIB_PREFIX}boost_${COMPONENT}-${_python_version}${_boost_MULTITHREADED}${_boost_RELEASE_ABI_TAG}-${Boost_LIB_VERSION}
         ${Boost_LIB_PREFIX}boost_${COMPONENT}-${_python_version}${_boost_MULTITHREADED}${_boost_RELEASE_ABI_TAG}
-        ${Boost_LIB_PREFIX}boost_${COMPONENT}-${_python_version} )
+        ${Boost_LIB_PREFIX}boost_${COMPONENT}-${_python_version}
+        ${Boost_LIB_PREFIX}boost_${COMPONENT}-${_python_major_version}${_boost_COMPILER}${_boost_MULTITHREADED}${_boost_RELEASE_ABI_TAG}-${Boost_LIB_VERSION}
+        ${Boost_LIB_PREFIX}boost_${COMPONENT}-${_python_major_version}${_boost_COMPILER}${_boost_MULTITHREADED}${_boost_RELEASE_ABI_TAG}
+        ${Boost_LIB_PREFIX}boost_${COMPONENT}-${_python_major_version}${_boost_MULTITHREADED}${_boost_RELEASE_ABI_TAG}-${Boost_LIB_VERSION}
+        ${Boost_LIB_PREFIX}boost_${COMPONENT}-${_python_major_version}${_boost_MULTITHREADED}${_boost_RELEASE_ABI_TAG}
+        ${Boost_LIB_PREFIX}boost_${COMPONENT}-${_python_major_version} )
     endif()
   if(_boost_STATIC_RUNTIME_WORKAROUND)
     set(_boost_RELEASE_STATIC_ABI_TAG "-s${_boost_RELEASE_ABI_TAG}")
