--- renpy/character.py
+++ renpy/character.py
@@ -672,7 +672,7 @@
             **self.show_args)
 
     # This is called after the last interaction is done.
-    def do_done(self, who, what):
+    def do_done(self, who, what, who_args=None):
         return
 
     # This is called when an extend occurs, before the usual add/show
@@ -826,7 +826,7 @@
             self.do_display(who, what, cb_args=self.cb_args, **display_args)
 
             # Indicate that we're done.
-            self.do_done(who, what)
+            self.do_done(who, what, self.who_args)
 
             # Finally, log this line of dialogue.
             if who and isinstance(who, (str, unicode)):
