diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f9e28170..bed46482 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -35,6 +35,9 @@
 SET(HEADER 
     client/BlockLocation.h
     client/DirectoryIterator.h
+    client/EncryptionZoneInfo.h
+    client/EncryptionZoneIterator.h
+    client/FileEncryptionInfo.h
     client/FileStatus.h
     client/FileSystem.h
     client/FileSystemStats.h
diff --git a/src/client/FileStatus.h b/src/client/FileStatus.h
--- a/src/client/FileStatus.h
+++ b/src/client/FileStatus.h
@@ -23,7 +23,7 @@
 #define _HDFS_LIBHDFS3_CLIENT_FILESTATUS_H_
 
 #include "Permission.h"
-#include "client/FileEncryptionInfo.h"
+#include "FileEncryptionInfo.h"
 
 #include <string.h>

