--- coregrind/m_aspacemgr/aspacemgr-linux.c	2010-10-21 01:24:09.000000000 +0400
+++ coregrind/m_aspacemgr/aspacemgr-linux.c	2011-08-09 04:53:51.980668244 +0400
@@ -265,13 +265,13 @@
 /* ------ start of STATE for the address-space manager ------ */
 
 /* Max number of segments we can track. */
-#define VG_N_SEGMENTS 5000
+#define VG_N_SEGMENTS 0x10000
 
 /* Max number of segment file names we can track. */
-#define VG_N_SEGNAMES 1000
+#define VG_N_SEGNAMES 0x1000
 
 /* Max length of a segment file name. */
-#define VG_MAX_SEGNAMELEN 1000
+#define VG_MAX_SEGNAMELEN 0x1000
 
 
 typedef
