diff --git a/projectmanagers/cmake/cmakecommitchangesjob.cpp b/projectmanagers/cmake/cmakecommitchangesjob.cpp
index e7b9d4e..7c1104c 100644
--- a/projectmanagers/cmake/cmakecommitchangesjob.cpp
+++ b/projectmanagers/cmake/cmakecommitchangesjob.cpp
@@ -135,8 +135,6 @@ void processDependencies(ProcessedTarget &target, const QString& dep, const CMak
 
     target.includes += depData["INTERFACE_INCLUDE_DIRECTORIES"];
     target.defines += depData["INTERFACE_COMPILE_DEFINITIONS"];
-    foreach(const QString& d, depData["INTERFACE_LINK_LIBRARIES"])
-        processDependencies(target, d, data);
 }
 
 Path::List CMakeCommitChangesJob::addProjectData(const CMakeProjectData& data)
