From 2893b2777d20357fc6154f828691e16cb71dbbe1 Mon Sep 17 00:00:00 2001
From: Vasiliy Yeremeyev <vayerx@gmail.com>
Date: Thu, 27 Jun 2013 16:53:39 +0400
Subject: [PATCH] initial amount of slugs: 8

---
 source/game/g_local.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/game/g_local.h b/source/game/g_local.h
index 224de33..722940d 100644
--- a/source/game/g_local.h
+++ b/source/game/g_local.h
@@ -1127,7 +1127,7 @@ extern void SP_misc_electroflash (edict_t *ent);
 #define BASE_ROCKETS	10
 #define BASE_SHELLS		10
 #define BASE_CELLS		50
-#define BASE_SLUGS		10
+#define BASE_SLUGS		8
 #define BASE_GRENADES	50
 #define BASE_BULLETS	50
 #define BASE_SEEKERS	1
-- 
1.8.1.5

